public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -24,23 +24,44 @@ interface VariableEditorFormProps {
   onTypeChange: (type: EditableVariableType) => void;
   onGoBack: () => void;
   onDelete: (variableName: string) => void;
+  onValidateVariableName: (name: string, key: string | undefined) => [true, string] | [false, null];
 }
-
-export function VariableEditorForm({ variable, onTypeChange, onGoBack, onDelete }: VariableEditorFormProps) {
+export function VariableEditorForm({
+  variable,
+  onTypeChange,
+  onGoBack,
+  onDelete,
+  onValidateVariableName,
+}: VariableEditorFormProps) {
   const styles = useStyles2(getStyles);
-  const { name, type, label, description, hide } = variable.useState();
+  const [nameError, setNameError] = useState<string | null>(null);
+  const { name, type, label, description, hide, key } = variable.useState();
   const EditorToRender = isEditableVariableType(type) ? getVariableEditor(type) : undefined;
   const [runQueryState, onRunQuery] = useAsyncFn(async () => {
     await lastValueFrom(variable.validateAndUpdate!());
   }, [variable]);
-
   const onVariableTypeChange = (option: SelectableValue<EditableVariableType>) => {
     if (option.value) {
       onTypeChange(option.value);
     }
   };
 
-  const onNameBlur = (e: FormEvent<HTMLInputElement>) => variable.setState({ name: e.currentTarget.value });
+  const onNameChange = useCallback(
```

Comment:
And here, we could mock `onValidateVariableName` to make sure it is called when the name input changes 😄